Python/optimizer.c

```diff
@@ -338,6 +339,13 @@ translate_bytecode_to_trace(
         ADD_TO_TRACE(SAVE_IP, (int)(instr - (_Py_CODEUNIT *)code->co_code_adaptive));
         int opcode = instr->op.code;
         uint64_t operand = instr->op.arg;
+        // TODO: EXTENDED_ARG handling
+        if (opcode == ENTER_EXECUTOR) {
+            _PyExecutorObject *executor = (_PyExecutorObject *)code->co_executors->executors[operand&255];
+            opcode = executor->vm_data.opcode;
+            DPRINTF(2, "  * ENTER_EXECUTOR -> %s\n",  _PyOpcode_OpName[opcode]);
+            operand = executor->vm_data.oparg;  // TODO: EXTENDED_ARG handling
+        }
         switch (opcode) {
             case LOAD_FAST_LOAD_FAST:
```

Comment:
We could do something where the code generator encodes this in the `size` field of the expansions, in the switch (currently) at line 405 below. E.g. `-1` means `operand >> 4` and `-2` means `operand & 0xF`. Should probably switch to an enum. The generator would have to work a little harder but that seems doable (e.g. it could detect the use of `oparg1` and `oparg2` and then parse the instruction name into two pieces).

Reply:
That should work, although I was thinking of the dumber approach of just annotating `LOAD_FAST_LOAD_FAST`, etc as a superinstruction in bytecodes.c. Something like adding `/* superinstruction */`